pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -169,6 +181,27 @@ private BrokerResponse handleRequest(long requestId, String query,
       return new BrokerResponseNative(QueryException.getException(QueryException.SQL_PARSING_ERROR, e));
     }
 
+    QueryPlan queryPlan = queryPlanResult.getQueryPlan();
+    Set<String> tableNames = getTableNamesFromRelRoot(queryPlanResult.getRelRoot());
+
+    // Compilation Time. This includes the time taken for parsing, compiling, create stage plans and assigning workers.
+    long compilationEndTimeNs = System.nanoTime();
+    long compilationTimeNs = (compilationEndTimeNs - compilationStartTimeNs) + sqlNodeAndOptions.getParseTimeNs();
+    updatePhaseTimingForTables(tableNames, BrokerQueryPhase.REQUEST_COMPILATION, compilationTimeNs);
+
+    // Validate table access.
+    if (!hasTableAccess(requesterIdentity, tableNames, requestId, requestContext)) {
+      return new BrokerResponseNative(QueryException.ACCESS_DENIED_ERROR);
+    }
+    updatePhaseTimingForTables(tableNames, BrokerQueryPhase.AUTHORIZATION, System.nanoTime() - compilationEndTimeNs);
+
+    // Validate QPS quota
+    if (hasExceededQPSQuota(tableNames, requestId, requestContext)) {
+      String errorMessage =
+          String.format("Request %d: %s exceeds query quota.", requestId, query);
```

Comment:
I see you're logging within the `hasExceededQPSQuota`. Do we need to log again here?